### ISSUE TITLE
Show type of constructors in patterns

### ISSUE THREAD

**Issue by DavsX** [commercialhaskell/intero#67]
Hi!
Given a file:

``` haskell
module ChapterExercises where

import Data.Monoid

data Two a b = Two a b  -- here it shows 'Two a b'

instance Foldable (Two a) where
  foldMap f (Two _ x) = f x -- here it shows 'Two a a'
```

The "Type of selection" on the data constructor Two shows 'Two :: forall a b. a -> b -> Two a b', which is correct. However if I press c-c c-t while the cursor is on Two in the last line, it shows 'Two :: Two a a', which is wrong.

I've never used Emacs before (trying to make it work with Vim), so if this is an user error, I aplogize :)

Edit: if I change the instance declaration line to 'Foldable (Two b)', it shows correctly 'Two b a'. So my guess is that it completes the missing types starting from 'a' even though 'a' is used already.

**Comment by chrisdone** [comment]
I believe that's because for pattern matches the boundary is the whole
pattern which is indeed of type Two a a. But it may be possible to extend
the info collection to include types of value constructors in there.

**Comment by chrisdone** [comment]
Fixed in #325 in 2016.
